CountOnMeTests/CalculatorModelTests.swift

```diff
@@ -0,0 +1,106 @@
+        self.calculatorModel = .init()
+        numberFormatter.locale = .current
+        numberFormatter.decimalSeparator = "."
+        numberFormatter.minimumFractionDigits = 4
+    }
+
+    func testStringSeparator() throws {
+        let rawString: String = "2 - 1 + 6 x 10 - 4 / 2 = "
+        let actualResult = calculatorModel.testSeparateStrings(rawString: rawString)
+        let expectedResult: [Substring] = ["2", "-", "1", "+", "6", "x", "10", "-", "4", "/", "2", "="]
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testRemovePrioritiesFromString() throws {
+        let rawString: String = "2 - 1 + 6 x 10 - 4 / 2 = "
+        calculatorModel.testApplyPriorityRules(rawString: rawString)
+        let expectedResult: [Substring] = ["2", "-", "1", "+", "60.0", "-", "2.0", "="]
+
+        XCTAssertEqual(calculatorModel.getStringResult(), expectedResult)
+    }
+
+    func testAddStringOperands() {
+        let leftOperand: Substring = .init("12")
+        let rightOperand: Substring = .init("2")
+        let actualResult = Operations.add(leftOperand, rightOperand)
+        let expectedResult: Substring = .init("14.0")
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testSubtractStringOperands() {
+        let leftOperand: Substring = .init("12")
+        let rightOperand: Substring = .init("2")
+        let actualResult = Operations.substract(leftOperand, rightOperand)
+        let expectedResult: Substring = .init("10.0")
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testMultiplyStringOperands() {
+        let leftOperand: Substring = .init("12")
+        let rightOperand: Substring = .init("2")
+        let actualResult = Operations.multiply(leftOperand, rightOperand)
+        let expectedResult: Substring = .init("24.0")
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testDivideStringOperands() {
+        let leftOperand: Substring = .init("12")
+        let rightOperand: Substring = .init("2")
+        let actualResult = Operations.divide(leftOperand, rightOperand)
+        let expectedResult: Substring = .init("6.0")
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testGetResultFromCalculator() {
+        let rawString: String = "2 x 3 + 1 - 10 / 5 = "
+        let expectedResult: String = "5.0"
+        let actualResult = calculatorModel.getResult(rawString: rawString)
+
+        XCTAssertEqual(actualResult, expectedResult)
+    }
+
+    func testGetResultFromCalculatorWithFormattedDecimalNumber() {
+        let rawString: String = "4 / 3 - 1 = "
+        let expectedResult: String = "0.3333"
+
+        let unformattedResult = calculatorModel.getResult(rawString: rawString)
+        guard let formattedResult = numberFormatter.string(from: NSDecimalNumber(string: unformattedResult)) else {
+            return
+        }
```

Comment:
**suggestion (blocking)**: If the formatting fails, the test will return as `passed` which is not the expected result. You can use `XCTFail(message: XXX)` to provide some context when something like this might happen.